src/main/java/seedu/address/model/book/BookName.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.model.book;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Book's name in LibTask.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBookName(String)}
+ */
+public class BookName {
+    public static final String MESSAGE_CONSTRAINTS =
```

Comment:
Might need to update the constraint message to allow ":" and other special characters.